### PR TITLE
Distance and closest point

### PR DESCRIPTION
**Pull request by Guarionex**
Addresses information between a shape and a point on issue #5

### CODE REVIEW

**Comment by 20chan** [comment]
Your all changes have formatting issues. This project uses [K&R indenting style](https://en.wikipedia.org/wiki/Indentation_style#K&R_style).

Can you change all of your changes? I'm really sorry for no .editorconfig or contributing docs. My VS2019 has problem with generating editorconfig file.

File: cirno/Vector.cs [hunk 1]
```diff
@@ -47,6 +47,13 @@ public struct Vector {
         public float Dot(Vector other) {
             return X * other.X + Y * other.Y;
         }
+        
+        public float Distance(Vector other)
+        {
+            var xPowDif = Math.Pow(other.X - X, 2);
+            var yPowDif = Math.Pow(other.Y - Y, 2);
+            return (float) Math.Sqrt(xPowDif + yPowDif);
```
Comment (20chan) [inline cirno/Vector.cs]: Can you remove whitespace after type casts?

File: cirno.Tests/Geometry/TestHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace cirno.Tests.Geometry
+{
+    public static class TestHelpers
+    {
+        public static float NextFloat(this Random random)
+        {
+            var buffer = new byte[4];
+            random.NextBytes(buffer);
+            return BitConverter.ToSingle(buffer,0);
```
Comment (20chan) [inline cirno.Tests/Geometry/TestHelpers.cs]: Can you add whitespace after comma?

File: cirno.Tests/Geometry/CircleTest.cs [hunk 3]
```diff
@@ -1,3 +1,5 @@
+using System;
```
Comment (20chan) [inline cirno.Tests/Geometry/CircleTest.cs:1]: Can you change ALL indent styles to K&R styles?

**Review by 20chan — CHANGES_REQUESTED** [review]

**Comment by Guarionex** [comment]
@20chan yeah I can make those changes later today

**Comment by Guarionex** [comment]
@20chan I've made the changes. Let me know if I missed one. Also the random float generator is kinda flaky due to overflow when squaring. I square root the random float to avoid overflow when squaring in the unit tests. I recommend you look into style cop as a linter to regulate code styles

File: cirno.Tests/Geometry/LineTest.cs [hunk 4]
```diff
@@ -29,66 +29,57 @@ public class LineTest {
...
-            
-            
+
             Assert.IsTrue(expected.Equals(actual, 0.01f));
         }
 
         [TestMethod]
-        public void GivenLine_WhenCallingDistanceToAPoint_ThenReturnDistance()
-        {
+        public void GivenLine_WhenCallingDistanceToAPoint_ThenReturnDistance(){
```
Comment (20chan) [inline cirno.Tests/Geometry/LineTest.cs]: Can you add whitespace before brace?

```csharp
method() {
```

**Review by 20chan — CHANGES_REQUESTED** [review]
Thank you for all these changes, only one more change left.

File: cirno.Tests/Geometry/LineTest.cs (hunk 4 above)
Comment (Guarionex) [reply cirno.Tests/Geometry/LineTest.cs]: Done

**Review by 20chan — APPROVED** [review]

File: cirno/Geometry/LineLike.cs [hunk 5]
```diff
@@ -41,5 +41,49 @@ public abstract class LineLike : IShape {
...
+            var bPointY = (P2.X - P1.X) * point.Y;
+            var c = P2.X * P1.Y - P2.Y * P1.X;
+
+            return Math.Abs(aPointX - bPointY + c) / lineDistance;
+        }
+
+        public Vector GetClosestPoint(Vector point) {
+            var closestPoint = new Vector();
+            var slope = GetSlopeOfLine();
+            if (float.IsInfinity(slope))
```
Comment (20chan) [inline cirno/Geometry/LineLike.cs:61]: Wait I missed this one.....

**Pull request merged**